app/src/main/res/layout/activity_main.xml

```diff
@@ -18,86 +18,89 @@
     For the additional permissions granted for Salt Edge Authenticator
     under Section 7 of the GNU General Public License see THIRD_PARTY_NOTICES.md
 -->
-<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
     android:id="@+id/activityRootLayout"
     android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:background="@android:color/white">
+    android:layout_height="match_parent">
 
-    <com.google.android.material.appbar.AppBarLayout
+    <androidx.constraintlayout.widget.ConstraintLayout
         android:id="@+id/appBarLayout"
         android:layout_width="match_parent"
-        android:layout_height="wrap_content">
+        android:layout_height="102dp"
+        android:background="@color/theme_background"
+        app:layout_constraintTop_toTopOf="parent">
 
         <FrameLayout
+            android:id="@+id/appBarTopSection"
+            android:orientation="horizontal"
             android:layout_width="match_parent"
-            android:layout_height="wrap_content">
+            android:layout_height="@dimen/dp_56"
+            android:gravity="center_vertical"
+            app:layout_constraintTop_toTopOf="parent">
 
-            <androidx.appcompat.widget.Toolbar
-                android:id="@+id/toolbarView"
-                android:layout_width="match_parent"
-                android:layout_height="@dimen/action_bar_size"
-                android:background="@color/color_primary"
-                android:theme="@style/ThemeOverlay.AppCompat.Dark.ActionBar" />
+            <ImageView
+                android:id="@+id/appBarBackAction"
+                android:layout_width="@dimen/dp_24"
+                android:layout_height="@dimen/dp_24"
+                android:layout_marginLeft="@dimen/dp_16"
+                android:layout_marginRight="@dimen/dp_16"
+                android:visibility="gone"
+                android:src="@drawable/ic_action_back"
+                />
         </FrameLayout>
-    </com.google.android.material.appbar.AppBarLayout>
 
-    <FrameLayout
-        android:id="@+id/container"
-        android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:layout_below="@+id/appBarLayout"
-        android:layout_above="@+id/bottomNavigationLayout"/>
+        <LinearLayout
```

Comment:
Can we use he functional of ConstraintLayout(chains)?

Reply:
Good idea but app bar is divided in two vertical sections and i should center vericaly this chain in the second container. I'll think about aligning to the center guidline.

Reply:
TODO or?